src/tree-view/treeViewComponent.ts

```diff
@@ -0,0 +1,81 @@
+const $ = (<any> window).$;
+const Promise = (<any> window).Promise;
+
+export class TreeViewController {
+  public tree;
+  public reselect;
+  public data;
+  public name : string;
+  public onSelect;
+  public lazyLoad;
+
+  /*@ngInject*/
+  constructor(private $element, private $timeout, private $http) {}
+
+  public $onInit() {
+    let element = this.$element[0].querySelector('div.treeview');
+    this.renderTree(element).then(() => {
+      this.tree = $(element).treeview(true);
+
+      this.tree.getNodes().forEach((node) => {
+        if (this.getTreeState(node) === !node.state.expanded) {
+          this.tree.toggleNodeExpanded(node);
+        }
+      });
+    });
+  }
+
+  private renderTree(element) {
+    return new Promise((resolve) => {
+      $(element).treeview({
+        data:            this.data,
+        expandIcon:      'fa fa-fw fa-angle-right',
+        collapseIcon:    'fa fa-fw fa-angle-down',
+        loadingIcon:     'fa fa-fw fa-spinner fa-pulse',
+        levels:          1,
+        allowReselect:   this.reselect,
+        showBorders:     false,
+        onNodeExpanded:  this.setTreeState(true),
+        onNodeCollapsed: this.setTreeState(false),
+        onNodeSelected:  (_event, node) => this.onSelect(node),
+        lazyLoad:        (node, render) => this.lazyLoad(node).then(render),
+        onRendered:      () => resolve()
+      });
+    });
+  }
+
+  private setTreeState(state) {
+    return (_event, node) => {
+      let persist = JSON.parse(sessionStorage.getItem('treeView-' + this.name));
```

Comment:
Instead of using concat use [Template_literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)
```typescript
let persist = JSON.parse(sessionStorage.getItem(`treeView-${this.name}`));
```